syntax = "proto3";
package tomcat;

message SliceDataPacket {
  bytes data = 1;
  uint32 col_count = 2;
  uint32 row_count = 3;
  uint64 timestamp = 4;
}

message VolumeDataPacket {
  bytes data = 1;
  uint32 col_count = 2;
  uint32 row_count = 3;
  uint32 slice_count = 4;
}

message ReconDataPacket {
  oneof data {
    VolumeDataPacket volume = 1;
    SliceDataPacket slice = 2;
  }
}

message SetSlicePacket {
  uint64 timestamp = 1;
  repeated float orientation = 2;
}

message ReconRequestPacket {
  oneof request {
    SetSlicePacket set_slice = 1;
  }
}